compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -567,14 +569,27 @@ pub(super) fn thir_abstract_const<'tcx>(
     }
 }
 
+/// Tries to unify two abstract constants using structural equality.
+#[instrument(skip(tcx), level = "debug")]
+pub(super) fn try_unify<'tcx>(
```

Comment:
why do we need this to be `pub(super)` here? additionally, why the `try_unify_inner` split, can't we use `ContextUnifyCtxt::new(tcx, param_env).try_unify(a, b)` everywhere instead?

Reply:
I thought this was the better abstraction, since always creating a `ConstUnifyCtxt` prior to that call didn't seem ideal at first. But we're calling `try_unify` only twice (besides the recursive calls) anyway and only in this file, so I got rid of the inner method.